// 语法版本
syntax = "proto3";

// package名
package grpc_demo.service;

// 服务: 可用于 RPC
// service 定义开放调用的服务，即 UserInfoService 微服务
service UserInfoService {
  // rpc 定义服务内的 GetUserInfo 远程调用
  rpc GetUserInfo (UserRequest) returns (UserResponse) {
  }
}

// 定义客户端请求的数据格式
message UserRequest {
  string name = 1;
}

// 定义服务端响应的数据格式
message UserResponse {
  string name = 1;
  int32 id = 2;
  int32 age = 3;
  string sex = 4;
} 
